src/js/utils/ui.js

```diff
@@ -201,6 +216,7 @@ function initInteractionListeners(ui) {
         // Always add this, in case we don't properly identify the device as mobile
         addEventListener(ui, initGroup, 'touchstart', interactStartHandler, listenerOptions);
     }
+    addEventListener(ui, initGroup, 'click', interactStartHandler);
```

Comment:
Given this code runs for all these types of event handlers, can we adjust this change so that we only add this "click" listener when UI is actually listening for click?

```
const eventRegisters = {
    drag(ui) {
        initInteractionListeners(ui);
    },
    dragStart(ui) {
        initInteractionListeners(ui);
    },
    dragEnd(ui) {
        initInteractionListeners(ui);
    },
    click(ui) {
        initInteractionListeners(ui);
    },
```

Otherwise, we are adding click listeners even to elements that we only want to listen to other events like drag or doubleclick.

Reply:
This additional listener causes `interactStartHandler` to fire on all UI instances on "click" and can result in these properties being modified between the expected mouse/pointer/touch handlers.
```
        ui.dragged = false;
        ui.lastStart = now();
        ui.startX = pageX;
        ui.startY = pageY;
```